End to End Fails on MacOS Catalina Due to Buffer Size Being Too Large in bufio.Scanner

<!--💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎

Hellooo! 😄 

To help us tend to your issue faster, please search our currently open issues before submitting a new one.
Existing issues often contain information about workarounds, resolution, or progress updates.

💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎💎-->

# 🐞 Bug Report

### Description

```
bazel test //endtoend:go_default_test --define=ssz=minimal --test_output=streamed --test_timeout=10000
```

Fails when running on MacOS catalina with the following error:

```
--- FAIL: TestEndToEnd_MinimalConfig (87.94s)
=== RUN   TestEndToEnd_Slashing_MinimalConfig
    TestEndToEnd_Slashing_MinimalConfig: endtoend_test.go:36: Shard index: 0
    TestEndToEnd_Slashing_MinimalConfig: endtoend_test.go:37: Starting time: 2020-09-15 18:59:19.377007 +0000 UTC m=+87.985860479
    TestEndToEnd_Slashing_MinimalConfig: endtoend_test.go:38: Log Path: /private/var/tmp/_bazel_johanna/8795d77cc667fd99d49d6319fba8c083/sandbox/darwin-sandbox/267/execroot/prysm/bazel-out/darwin-fastbuild/testlogs/endtoend/go_default_test/test.outputs
    TestEndToEnd_Slashing_MinimalConfig: eth1.go:73: mining log not found, this means the eth1 chain had issues starting: bufio.Scanner: token too long
--- FAIL: TestEndToEnd_Slashing_MinimalConfig (2.01s)
FAIL
FAIL: //endtoend:go_default_test (see /private/var/tmp/_bazel_johanna/8795d77cc667fd99d49d6319fba8c083/execroot/prysm/bazel-out/darwin-fastbuild/testlogs/endtoend/go_default_test/test.log)
Target //endtoend:go_default_test up-to-date:
  bazel-bin/endtoend/darwin_amd64_stripped/go_default_test
INFO: Elapsed time: 100.176s, Critical Path: 99.46s
INFO: 68 processes: 68 darwin-sandbox.
INFO: Build completed, 1 test FAILED, 72 total actions
//endtoend:go_default_test                                               FAILED in 90.7s
  /private/var/tmp/_bazel_johanna/8795d77cc667fd99d49d6319fba8c083/execroot/prysm/bazel-out/darwin-fastbuild/testlogs/endtoend/go_default_test/test.log
INFO: Build completed, 1 test FAILED, 72 total actions
```

### Has this worked before in a previous version?

Yes, it worked in catalina a while back, but not sure which commit

## 🔬 Minimal Reproduction

```
bazel test //endtoend:go_default_test --define=ssz=minimal --test_output=streamed --test_timeout=10000
```

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
MacOSX Catalina
  </code>
</pre>

**What version of Prysm are you running? (Which release)**

<pre>
  <code>
Alpha.25
  </code>
</pre>

--- a/endtoend/helpers/helpers.go
+++ b/endtoend/helpers/helpers.go
@@ -25,6 +25,8 @@ const (
 	filePollingInterval = 500 * time.Millisecond
 	memoryHeapFileName  = "node_heap_%d.pb.gz"
 	cpuProfileFileName  = "node_cpu_profile_%d.pb.gz"
+	fileBufferSize      = 64 * 1024
+	maxFileBufferSize   = 1024 * 1024
 )
 
 // KillProcesses finds the passed in process IDs and kills the process.
@@ -79,8 +81,8 @@ func WaitForTextInFile(file *os.File, text string) error {
 			return fmt.Errorf("could not find requested text \"%s\" in logs:\n%s", text, contents)
 		case <-ticker.C:
 			fileScanner := bufio.NewScanner(file)
-			buf := make([]byte, 0, 64*1024)
-			fileScanner.Buffer(buf, 1024*1024)
+			buf := make([]byte, 0, fileBufferSize)
+			fileScanner.Buffer(buf, maxFileBufferSize)
 			for fileScanner.Scan() {
 				scanned := fileScanner.Text()
 				if strings.Contains(scanned, text) {

--- a/endtoend/helpers/helpers.go
+++ b/endtoend/helpers/helpers.go
@@ -78,18 +78,15 @@ func WaitForTextInFile(file *os.File, text string) error {
 			}
 			return fmt.Errorf("could not find requested text \"%s\" in logs:\n%s", text, contents)
 		case <-ticker.C:
-			fileScanner := bufio.NewScanner(file)
-			for fileScanner.Scan() {
-				scanned := fileScanner.Text()
-				if strings.Contains(scanned, text) {
+			r := bufio.NewReader(file)
+			line, err := r.ReadString('\n')
+			for err == nil {
+				if strings.Contains(line, text) {
 					return nil
 				}
+				line, err = r.ReadString('\n')
 			}
-			if err := fileScanner.Err(); err != nil {
-				return err
-			}
-			_, err := file.Seek(0, io.SeekStart)
-			if err != nil {
+			if err != io.EOF {
 				return err
 			}
 		}

--- a/endtoend/helpers/helpers.go
+++ b/endtoend/helpers/helpers.go
@@ -78,15 +78,20 @@ func WaitForTextInFile(file *os.File, text string) error {
 			}
 			return fmt.Errorf("could not find requested text \"%s\" in logs:\n%s", text, contents)
 		case <-ticker.C:
-			r := bufio.NewReader(file)
-			line, err := r.ReadString('\n')
-			for err == nil {
-				if strings.Contains(line, text) {
+			fileScanner := bufio.NewScanner(file)
+			buf := make([]byte, 0, 64*1024)
+			fileScanner.Buffer(buf, 1024*1024)
+			for fileScanner.Scan() {
+				scanned := fileScanner.Text()
+				if strings.Contains(scanned, text) {
 					return nil
 				}
-				line, err = r.ReadString('\n')
 			}
-			if err != io.EOF {
+			if err := fileScanner.Err(); err != nil {
+				return err
+			}
+			_, err := file.Seek(0, io.SeekStart)
+			if err != nil {
 				return err
 			}
 		}

